license-generator/generate.hs

```diff
@@ -59,9 +59,14 @@ depsNamesAndVersions :: IO [(String, String)]
 depsNamesAndVersions = do
   contents <- lines <$> getContents
   deps <- traverse parse contents
-  pure (filter (\(name, _) -> name /= "purescript" && name /= "rts") deps)
+  pure (filter (\(name, _) -> not (excluded name)) deps)
 
   where
+  excluded name =
+    name == "purescript"
+    || name == "purescript-ast"
+    || name == "rts"
```

Comment:
Is this the right thing to do? I assume we were ignoring `purescript` because we can't look up our license over the wire while we're trying to build out license. I assume we were ignoring `rts` because it's not a real hackage package.

Since `purescript-ast` is a package we own, we would run into a similar issue as the `purescript` package trying to lookup its own license on hackage`. But, since `purescript` depends on `purescript-ast`, shouldn't it have the `purescript-ast` as part of the license? Should `purescript-ast` not define its own license and instead use the top-level `LICENSE` file? I don't have an answer to any of these questions. I'm down to change to whatever.

Reply:
`purescript-ast` should definitely have its own LICENSE file, yeah. Since we won't be distributing it as a binary, just as source code, we won't be distributing it alongside its dependencies, so we needn't bother with a license-generator setup like we have with the compiler now. So I'd say the LICENSE files for any split-off packages should just be the normal BSD3 license text and nothing else.

Reply:
Okay, that makes sense. Given that information, I'd assume the license generator should be changed to pull in the license from local packages that are used. That sound right?

Reply:
Oh yeah that’s a good point, I previously assumed that they would be included in the output of `stack ls dependencies` but I guess that is probably incorrect

Reply:
The packages are in there, but we look for the licenses from `hackage` and the local packages don't have licenses on hackage.

Reply:
Having written that out, I guess the question is, when should this license generator be run? Is it expected to be run only when making a new release (in which case, the local packages should be on `hackage`)? Or should it always be possible to run the license generator (in which case, we should handle local packages differently)?

Reply:
I'm going with the idea that we only care about running `make license-generator` when doing a release (so local packages should be on `hackage`) based on the [release guide](https://github.com/purescript/purescript/blob/fcc7a08a21a8de1dafb8624de02bfc94f81341b9/RELEASE_GUIDE.md#before-making-a-release).

Lemme know if that's the wrong decision.